### ISSUE TITLE
Incorrect error span on classes with both instance and static index signatures

### ISSUE THREAD

**Issue by DanielRosenwasser** [microsoft/TypeScript#44134]
```ts
class Foo {
  [p: string]: any;
  static [p: string]: number;
}
```

```
error: Property 'prototype' of type 'Foo' is not assignable to string index type 'number'.
```
